fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -3853,15 +3974,175 @@ int DDTeamCollection::overlappingMachineMembers(std::vector<Standalone<StringRef
 	return maxMatchingServers;
 }
 
+void DDTeamCollection::cleanupLargeTeams() {
+	for (int t = 0; t < std::min<int>(largeTeams.size(), SERVER_KNOBS->DD_MAXIMUM_LARGE_TEAM_CLEANUP); t++) {
+		if (!shardsAffectedByTeamFailure->hasShards(
+		        ShardsAffectedByTeamFailure::Team(largeTeams[t]->getServerIDs(), primary))) {
+			largeTeams[t]->tracker.cancel();
+			largeTeams[t--] = largeTeams.back();
+			largeTeams.pop_back();
+		}
+	}
+}
+
+int DDTeamCollection::maxLargeTeamSize() const {
+	std::vector<Reference<TCServerInfo>> healthy;
+	for (auto& [serverID, server] : server_info) {
+		if (!server_status.get(serverID).isUnhealthy()) {
+			healthy.push_back(server);
+		}
+	}
+
+	if (!satisfiesPolicy(healthy)) {
+		return -1;
+	}
+	return healthy.size();
+}
+
+struct ServerPriority {
+	int healthyUsed = 0;
+	int unhealthyUsed = 0;
```

Comment:
It might be good to clarify that we're tracking the number of shards for this server. Maybe we could call these `healthyShards` / `unhealthyShards` instead? But that's also a bit confusing because the shards themselves aren't healthy/unhealthy.